server/protocols/webadmin/webadmin-rabbitmq/src/main/java/org/apache/james/webadmin/routes/RabbitMQRoutes.java

```diff
@@ -0,0 +1,165 @@
+import spark.Service;
+
+@Api(tags = "MailQueues")
+@Path(RabbitMQRoutes.BASE_URL)
+@Produces("application/json")
+public class RabbitMQRoutes implements Routes {
+
+    public static final String BASE_URL = "/mailQueues";
+
+    private final MailQueueFactory<? extends ManageableMailQueue> mailQueueFactory;
+    private final JsonTransformer jsonTransformer;
+    private final TaskManager taskManager;
+    private final Clock clock;
+
+    @Inject
+    @SuppressWarnings("unchecked")
+    @VisibleForTesting
+    RabbitMQRoutes(MailQueueFactory<? extends ManageableMailQueue> mailQueueFactory,
+                   Clock clock, JsonTransformer jsonTransformer, TaskManager taskManager) {
+        this.mailQueueFactory = mailQueueFactory;
+        this.clock = clock;
+        this.jsonTransformer = jsonTransformer;
+        this.taskManager = taskManager;
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_URL;
+    }
+
+    @Override
+    public void define(Service service) {
+        rebuildMailQueue(service);
+    }
+
+
+    @POST
+    @Path("/spooler")
+    @ApiOperation(
+        value = "rebuild the RabbitMQ MailQueue using the cassandra mail queue view"
+    )
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "OK, the task for rebuilding the queue is created"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid request for rebuilding the mail queue."),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    public void rebuildMailQueue(Service service) {
+        TaskFromRequest taskFromRequest = this::rebuildMailQueue;
+        service.post(BASE_URL + "/spooler",
+            taskFromRequest.asRoute(taskManager),
+            jsonTransformer);
+    }
+
+    private Task rebuildMailQueue(Request request) {
+        checkTask(request, "rebuild");
+        return new RebuildMailQueueTask(getSpoolerMailQueue(), getOlderThan(request));
+    }
+
+    private RabbitMQMailQueue getSpoolerMailQueue() {
+        ManageableMailQueue fetchedMailQueue = mailQueueFactory.getQueue(MailQueueFactory.SPOOL)
+            .orElseThrow(() -> ErrorResponder.builder()
+                .message("Spooler MailQueue not created")
+                .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                .type(ErrorResponder.ErrorType.SERVER_ERROR)
+                .haltError());
+
+        if (!(fetchedMailQueue instanceof RabbitMQMailQueue)) {
+            throw ErrorResponder.builder()
+                .message("Spooler MailQueue is not a RabbitMQ queue")
+                .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                .type(ErrorResponder.ErrorType.SERVER_ERROR)
+                .haltError();
+        }
+
+        return (RabbitMQMailQueue) fetchedMailQueue;
+    }
+
+    private void checkTask(Request request, String expectedTaskName) {
```

Comment:
Don't do that yourself!

You can rely on 

```
TaskFromRequestRegistry.builder()
    .parameterName("task")
    .registration(this::rebuildMailQueue)
    .build();
```

Reply:
It looks obscure and it does not check its actual value.

Reply:
i think that it obfuscate what we want to do but convince me :)

Reply:
All other webadmin tasks rely on this mechanism.

I am against doing something different here.

Reply:
we did not find any example with an already existing route, if it is doable, maybe we can mob on this tomorrow?

Reply:
I managed to make it work, against all odds